app/components/windows/WidgetEditor.vue

```diff
@@ -0,0 +1,367 @@
+            <span class="subsection__title">{{ slot.label }}</span>
+            <div class="subsection__content"><slot :name="slot.value" /></div>
+          </div>
+          <div class="subsection">
+            <span class="subsection__title">{{ $t('Sources and Settings') }}</span>
+            <ul style="margin: 0;">
+              <li
+                class="subsection__content settings-title"
+                v-for="setting in settings"
+                :class="{ active: currentSetting === setting.value }"
+                :key="setting.value"
+                @click="updateCurrentSetting(setting.value)"
+              >{{ setting.label }}</li>
+            </ul>
+          </div>
+          <div class="subsection">
+            <span class="subsection__title">{{ $t('Selected Properties') }}</span>
+            <div class="subsection__content" v-if="currentSetting !== 'source'">
+              <slot :name="`${currentSetting}-properties`" v-if="!loadingFailed"/>
+              <div v-else>
+                {{ $t('Failed to load settings') }}
+              </div>
+            </div>
+            <div class="subsection__content" v-if="currentSetting === 'source'">
+              <generic-form v-model="sourceProperties" @input="onPropsInputHandler"/>
+            </div>
+          </div>
+        </div>
+
+        <div class="code-editor hidden" ref="code">
+          <tabs
+            :showContent="false"
+            className="widget-editor__top-tabs"
+            :tabs="codeTabs"
+            @input="value => updateCodeTab(value)"
+            :value="currentCodeTab"
+          />
+          <div v-for="tab in codeTabs" :key="tab.value">
+            <slot :name="tab.value" v-if="tab.value === currentCodeTab" />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</modal-layout>
+</template>
+
+<script lang="ts" src="./WidgetEditor.vue.ts"></script>
+
+<style lang="less">
+  @import "../../styles/index";
+
+  .widget-editor__top-tabs {
+    height: 36px !important;
+    width: 100%;
+
+    .tab-button {
+      text-transform: uppercase;
+      height: 36px;
+      font-size: 12px;
+      position: relative;
+      bottom: -1px;
+      z-index: 1;
+    }
+  }
+
+  .night-theme {
+    .widget-editor__top-tabs {
+      background-color: @night-section !important;
+      border-bottom: 1px solid @night-editor-border !important;
+    }
+  }
+
+  .top-settings {
+    .input-container {
+      margin-bottom: 0;
+      width: auto;
+      flex-direction: column;
+      margin-right: 8px;
+      margin-bottom: 16px;
+    }
+    .int-input{
+      width: 60px;
+    }
+    input[type=text] {
+      height: 28px;
+    }
+  }
+
+  .subsection__content {
+    .input-wrapper {
+      width: 100%;
+    }
+    .input-label {
+      width: 0;
+      padding: 0;
+    }
+  }
+
+  .test-button .link {
+    color: @white;
+  }
+</style>
+
+<style lang="less" scoped>
```

Comment:
What is the reason for having 2 style blocks here?

Reply:
one is unscoped